### ISSUE TITLE
how to use on android?

### ISSUE THREAD

**Issue by 0x410c** [alexer/offsetof-tool#1]
i am trying to use the tool for android..

```
Makefile:9: recipe for target 'default' failed
subprocess.CalledProcessError: Command '('make',)' returned non-zero exit status 2.
```

I am doing this on windows linux subsystem and giving header file of android source as input?
any idea whats wrong?

p.s: thank you for the tool!

**Comment by alexer** [comment]
I haven't done any Android development (...some day...) - much less any Android kernel development - nor worked with WSL, so I have no idea what kind of a setup that is... but the `--kernel` part of the tool was made to work on Linux, against the currently running kernel, so it isn't surprising it doesn't work. :P

(Your error message also seems to be missing quite a bit of stuff, not the least of which is the actual reason that `make` failed - which would probably be "No such file or directory")

If you're not targeting the currently running kernel, the bare minimum you need to do is change `KERNELDIR ?= /lib/modules/$(shell uname -r)/build` in the Makefile (which is embedded in offsetof.py) to the actual path to the desired kernel source. Since you're also cross-compiling the kernel, based on a cursory search you probably also need to add the `ARCH=arm` and `CROSS_COMPILE=/path/to/cross-compiler` arguments to the `make` invocation inside the embedded Makefile.

So basically, to get it to work, you should first figure out what's the proper way to do an out-of-tree kernel module build for android in your environment, and then modify the embedded Makefile accordingly. With any luck, the two changes I mentioned might be all that you need - but like I said, I haven't really worked with any of the components that you're working with, so...

**Comment by 0x410c** [comment]
great! thats what i was thinking,

thanks for the reply!

P.S: good work!
